### PR TITLE
pmemcheck: DEFAULT_REORDER support

### PR DESCRIPTION
**Pull request by wlemkows**
This change is [Reviewable](https://reviewable.io/reviews/pmem/valgrind/56)

### CODE REVIEW

File: pmemcheck/pmemcheck.h [hunk 1]
```diff
@@ -79,7 +79,8 @@ typedef
        VG_USERREQ__PMC_REMOVE_FROM_TX_N,
        VG_USERREQ__PMC_ADD_THREAD_TO_TX_N,
        VG_USERREQ__PMC_REMOVE_THREAD_FROM_TX_N,
-       VG_USERREQ__PMC_ADD_TO_GLOBAL_TX_IGNORE
+       VG_USERREQ__PMC_ADD_TO_GLOBAL_TX_IGNORE,
+       VG_USERREQ__PMC_DEFAULT_REORDER
```
Comment (marcinslusarz) [inline pmemcheck/pmemcheck.h]: comma on the last line - you won't need to modify this line when someone will add new request

File: pmemcheck/docs/pmc-manual.xml [hunk 2]
```diff
@@ -706,6 +706,18 @@ Overlapping regions:
         </listitem>
       </varlistentry>
 
+      <varlistentry id="crm.default_reorder" xreflabel="VALGRIND_PMC_DEFAULT_REORDER">
+        <term>
+          <option><![CDATA[VALGRIND_PMC_DEFAULT_REORDER]]></option>
+        </term>
+        <listitem>
+          <para>
+            Issues a default reorder engine. Useful in some types of
```
Comment (marcinslusarz) [inline pmemcheck/docs/pmc-manual.xml]: "default reorder engine **event**" or something like that

**Comment by wlemkows** [comment]
Review status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @marcinslusarz)

---

*[pmemcheck/pmemcheck.h, line 83 at r1](https://reviewable.io/reviews/pmem/valgrind/56#-LFpsXhOE1dLhleDpYks-r1-83:-LFqL6V77plFy4Ol5Vpx:b-896fix) ([raw file](https://github.com/pmem/valgrind/blob/883aea03994a47b176fa40dce74fe22f4cc4518c/pmemcheck/pmemcheck.h#L83)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

comma on the last line - you won't need to modify this line when someone will add new request
</blockquote></details>

Done.

---

*[pmemcheck/docs/pmc-manual.xml, line 715 at r1](https://reviewable.io/reviews/pmem/valgrind/56#-LFpsXhOE1dLhleDpYkq-r1-715:-LFqL7J6BllMNvvZl_VK:b-896fix) ([raw file](https://github.com/pmem/valgrind/blob/883aea03994a47b176fa40dce74fe22f4cc4518c/pmemcheck/docs/pmc-manual.xml#L715)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

"default reorder engine **event**" or something like that
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/pmem/valgrind/56)*

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 2 of 4 files at r1, 3 of 3 files at r2.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/pmem/valgrind/56#-:-LFqLs8U8xjIZfXpNNEe:bnfp4nl)*

**Pull request merged**